presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java

```diff
@@ -37,10 +37,11 @@ protected boolean supportsNotNullColumns()
         return false;
     }
 
+    // Hive connector currently does not support row-by-row delete
     @Override
-    public void testDelete()
+    protected boolean supportsRowByRowDelete()
     {
-        // Hive connector currently does not support row-by-row delete
+        return false;
```

Comment:
This looks "right", but also confuses me. Can you help me understand why you are doing this?

Reply:
This approach is more correct since it doesn't silently hide what is going on.  We push some logic to AbstractTestDistributedQueries (ie which tests require what), which is actually where it belongs, since that is where the queries live.  All we should know in this Test is what we support, not which tests require that support (since we don't see the queries here).  There are three additional reasons why we are doing this:

Firstly:  the test is now Omitted and doesn't simply pass, so we aren't deceiving the person running the test.

Secondly: it reduces duplicate code and duplicate reasoning, since this test needs to be omitted in both this file and the file we added.  If row-by-row deletion ever doesn't become a requirement for 'testDelete()', or Hive connector begins to support it at some point in the future, you have a more robust system to un-omit the test.  Specifically, our reasoning for omitting the test is recorded by our structure and not our comment.

Thirdly: this avoids having two different ways of Omitting tests within this file.

Reply:
Makes sense. But see my comment below